astropy/utils/masked/function_helpers.py

```diff
@@ -1433,6 +1433,145 @@ def setdiff1d(ar1, ar2, assume_unique=False):
     return ar1[np.isin(ar1, ar2, assume_unique=True, invert=True)], None, None
 
 
+@dispatched_function
+def einsum(*operands, out=None, **kwargs):
+    from astropy.utils.masked import Masked, MaskedNDArray
+
+    # input validation
+    subscripts, *operands = operands
+
+    if not isinstance(subscripts, str):
+        raise ValueError('only "subscripts" string mode supported for einsum.')
+
+    kwargs_with_out = kwargs.copy()
+    if out is not None:
+        if not isinstance(out, MaskedNDArray):
+            raise NotImplementedError
+
+        else:
+            kwargs_with_out["out"] = out.unmasked.view(np.ndarray)
```

Comment:
Why the `.view(np.ndarray)`? If this is, say, a `Quantity`, then I think it should still be dispatched on that.